paasta_tools/config_utils.py

```diff
@@ -180,7 +180,13 @@ def __enter__(self):
         self.pwd = os.getcwd()
         os.chdir(self.working_dir)
         if self.branch != "master":
-            subprocess.check_call(["git", "checkout", "-b", self.branch])
+            try:
+                # If there is a remote branch, checkout that branch
+                subprocess.check_call(
+                    ["git", "checkout", "-b", self.branch, f"origin/{self.branch}"]
+                )
+            except subprocess.CalledProcessError:
+                subprocess.check_call(["git", "checkout", "-b", self.branch])
```

Comment:
i think ideally we'd check if the remote branch exists using something like `ls-remote` rather than assuming that this checkout failing means there's no remote branch (since we're unconditionally assuming that this function will only ever error when the remote doesn't exist) - but this flag is really only for testing purposes so it's probably not worth the effort/complexity to change :)